### ISSUE TITLE
Internationalization and localization (i18n)

### ISSUE THREAD

**Issue by cdelahousse** [SteamedPears/Code-Review#48]
We should consider internationalizing Code Review. Many peeps don't speak english.

**Comment by cdelahousse** [comment]
http://ejohn.org/blog/a-strategy-for-i18n-and-node/
http://ejohn.org/blog/i18n-module-for-node-and-express-js/

**Comment by psimonyi** [comment]
One thing I often do is put localizable text in double quotes, and other
text in singles. I figure that way, I don't have to do much to find them
when I eventually put in a localization library.

**Comment by spratt** [comment]
I don't think we have a lot of text.  I like simple interfaces.  That said, I have no idea what text isn't localizable.  Obviously we wouldn't want to localize our brand, so CodeReview should be CodeReview everywhere.  But all the other text is just interface.

**Comment by spratt** [comment]
Since node isn't serving any of the interface, the node internationalization libraries won't help.  We would want something client side to pull down the appropriate text for the chosen language.

**Comment by spratt** [comment]
It should be easy enough to store the text in a json file like [{"en":"Submit","fr":"Submissez",...},{"en":"Code goes here","fr":"Code va içi",...},...]

Then we could load in the json file, and write the string to the dom.

**Comment by bheesham** [comment]
Separate the languages by files.

You could have a attribute tag called "data-i18n", which points to the language key. But that brings up the problem of if they don't have JS enabled. And if they don't, the textarea will look stupid, and nothing would have text. Is that something we'd want to worry about?

**Comment by psimonyi** [comment]
I'm pretty sure our application is totally unusable without JavaScript.
If that is the case, then it doesn't matter. (Or we should work on
fixing that; either one.)

**Comment by spratt** [comment]
For now, let's require javascript.  Otherwise would require generating pages on the backend, which is something I'd really like to avoid.

**Comment by bheesham** [comment]
Heh, then I guess we could add the text after.

**Comment by psimonyi** [comment]
@spratt It isn't obvious that we wouldn't want to localize our brand. It may have unfortunate associations in other languages, or be terribly unpronounceable, etc. Or it might just look better slightly translated: fr.wikipedia.org is Wikipédia. I don't think we should dismiss brand translation so lightly if we're considering translating other stuff.
